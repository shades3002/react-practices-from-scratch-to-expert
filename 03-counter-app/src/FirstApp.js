import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

// const FirstApp = ( props ) => {  console.log(props.hi) }
// Funtional Component
const FirstApp = ( { hi = 'hi', subtitle } ) => {

    const hello = 'Hello World';
    const objHello = {
        name: 'Carlos',
        color: 'Blue'
    }

    console.log(hi);

    // return (
    //     <div>
    //         <h1>Hello World</h1>
    //         <p>My first application</p>
    //     </div>
    // );

    return (
        <Fragment>
            <h1> { hello } </h1>
            <pre>{ JSON.stringify(objHello, null, 3) }</pre>
            <h6> { [1, 6, 9, 10] } </h6>
            {/* <h5> { 5 + 6 } </h5> */}
            <p>My first application</p>
            <p>{ hi }</p>
            <p>{ subtitle }</p>
        </Fragment>
    );

    // short Fragment
    // return (
    //     <>
    //         <h1>Hello World</h1>
    //         <p>My first application</p>
    //     </>
    // );
}

FirstApp.propTypes = {
    hi: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitle: 'I am subtitle'
}

export default FirstApp;