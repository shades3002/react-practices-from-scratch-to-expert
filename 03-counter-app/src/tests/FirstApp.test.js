import React from 'react';
// import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import FirstApp from "../FirstApp";



describe('Test component <FirstApp />', ()=> {

    // test('should show hi', () => {
    //     const hi = 'hi';
    //     const { getByText } = render( <FirstApp hi = { hi } /> );
    //     expect( getByText( hi ) ).toBeInTheDocument();
    // });

    test('should show <FirstApp /> correctly', () => {
        const hi = 'Hi';
        const wrapper = shallow( <FirstApp hi={ hi } /> );

        expect( wrapper ).toMatchSnapshot();
    });
    
    test('should subtitle correctly', () => {
        const hi = 'Hi';
        const subtitle = 'I am subtitle';
        const wrapper = shallow( 
            <FirstApp 
                hi={ hi } 
                subtitle={ subtitle }
            /> 
        );

        const textSubtitle = wrapper.find('p ~ p ~ p').text();

        expect( textSubtitle ).toBe( subtitle );
    });

});