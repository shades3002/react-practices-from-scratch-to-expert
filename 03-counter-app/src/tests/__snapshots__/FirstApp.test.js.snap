// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <FirstApp /> should show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
     
    Hello World
     
  </h1>
  <pre>
    {
   "name": "Carlos",
   "color": "Blue"
}
  </pre>
  <h6>
     
    1
    6
    9
    10
     
  </h6>
  <p>
    My first application
  </p>
  <p>
    Hi
  </p>
  <p>
    I am subtitle
  </p>
</Fragment>
`;

exports[`Test component <FirstApp /> should subtitle correctly 1`] = `
<Fragment>
  <h1>
     
    Hello World
     
  </h1>
  <pre>
    {
   "name": "Carlos",
   "color": "Blue"
}
  </pre>
  <h6>
     
    1
    6
    9
    10
     
  </h6>
  <p>
    My first application
  </p>
  <p>
    Hi
  </p>
  <p>
    I am subtitle
  </p>
</Fragment>
`;
